### ISSUE TITLE
LO Freq Offset shift?

### ISSUE THREAD

**Issue by majortom9** [g4ixt/QtTinySA#35]
Hi,
would you be able to implement a checkbox or something to integrate an LO offset, in order to shift the displayed frequency
by a user defined LO Frequency Offset? 

See below commit I added to do this (minus the necessary UI mods) for my limited use case. eg - looking at the IF output signal from a CBAND Satellite Antenna LNB output, this suits me.

https://github.com/majortom9/QtTinySA/commit/57145365022fb781325288889650f1548928f4e5

Here is an example display, for an input spectrum of 950-1150 MHz. This way it's displayed for me in terms of the
actual over the air frequency. I would have given it a go myself, but I am not familiar with PyQt development.
Have had limited experience with Qt / C++ though.
IF output = 5150 MHz - Sat Freq (OTA).

<a href="https://ibb.co/j6yNpWX"><img src="https://i.ibb.co/LkxFGJ7/Qt-Tiny-SA-v0-9-1-modified.png" alt="Qt-Tiny-SA-v0-9-1-modified" border="0"></a>

**Comment by g4ixt** [comment]
Yes, I like that idea it will be useful for AO-100.  I have just modified the v0.9.3 config database tables to allow for this enhancement in future (a checkbox and spinbox).  Depending on how easy it is to code I might add it into the next 'release' but I have almost finished working on that.

**Comment by majortom9** [comment]
that would be awesome, thanks. btw, in case I wanna play around, follow along...
what IDE tool(s) are you using to develop on? No windows machines, so I am limited to
linux tools here.

**Comment by g4ixt** [comment]
I haven't been able to add it to the version I was working on because (compared with the version you modified) I have made a lot of changes to the markers and it wasn't easy to replicate your changes without breaking things.  I will do it for the next version.  I have added a couple of things though:  I modified the config database, ready, and added a module called 'freqOffset' in the display Class, and in the preferences dialogue I have put a LO freq spinbox and radio buttons for high/low mixing.  None of this does anything yet.

I just pushed this latest version v0.10.0 to my development branch and my next jobs are to build and test executables and then deploy it to the main branch.

I use Anaconda with the Spyder 5.15.10 Python IDE and Qt Designer 5.15.2 for the GUI.  Python version is 3.11.7.  I use pyinstaller for building the executables.  All running in Kubuntu Linux 22.04 LTS.

I use Windows 11 just for building and (rudimentary) testing the Windows .exe

**Comment by majortom9** [comment]
I looked in to the database not saving the Lo Freq, and it looks like this patch on top of the one above will fix that.

`qttinysa]$ git diff
diff --git a/QtTinySA.py b/QtTinySA.py
index d4cd013..7b96d08 100755
--- a/QtTinySA.py
+++ b/QtTinySA.py
@@ -654,7 +654,6 @@ class display:
     def updateTrace(self, frequencies, readings):  # called by sigProcess() for every trace every 20 points
         adjusted_frequencies = self.freqOffset(frequencies)
         self.trace.setData((adjusted_frequencies/1e6), readings)
-        #self.trace.setData((frequencies/1e6), readings)
         if ui.grid.isChecked():
             tinySA.vGrid.show()
         else:
@@ -1301,6 +1300,7 @@ numbers.dwm.addMapping(preferences.maxPoints, 4)
 numbers.dwm.addMapping(ui.start_freq, 5)
 numbers.dwm.addMapping(ui.stop_freq, 6)
 numbers.dwm.addMapping(preferences.peakThreshold, 7)
+numbers.dwm.addMapping(preferences.freqLO, 8)
 numbers.tm.select()
 numbers.dwm.setCurrentIndex(0)
 markers.createTableModel()
`

**Comment by majortom9** [comment]
I was able to modify the QtTSAprefs.db, 'numbers' table structure, using sqlitebrowser, and save the highLO setting status to the database.
Now these three patches on top of your last commit, 06738c7e from 'main' branch are working for me. 
Take a look and see what ya think.  

[0003-save-external-LO-high-low-setting-to-the-database.patch.txt](https://github.com/g4ixt/QtTinySA/files/14546950/0003-save-external-LO-high-low-setting-to-the-database.patch.txt)
[0002-save-freqLO-to-database.patch.txt](https://github.com/g4ixt/QtTinySA/files/14546951/0002-save-freqLO-to-database.patch.txt)
[0001-use-LO-Offset-for-a-user-defined-external-LO.patch.txt](https://github.com/g4ixt/QtTinySA/files/14546952/0001-use-LO-Offset-for-a-user-defined-external-LO.patch.txt)

**Comment by g4ixt** [comment]
I will have a look at your patches.  I have been having a bit of a break after spending a lot of time on the 0.10.1 version and was just going to have a look at it again.

**Comment by g4ixt** [comment]
Implemented in v0.10.2
